File: courtlistener/api/opinions_cited.py
"""
Opinions Cited API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class OpinionsCitedAPI(BaseAPI):
    """API for accessing citation relationships between opinions."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "opinions-cited/"
    
    def list(
        self,
        citing_opinion: Optional[int] = None,
        cited_opinion: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List citation relationships.
        
        Args:
            citing_opinion: Filter by citing opinion ID
            cited_opinion: Filter by cited opinion ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of citation relationships
        """
        params = {}
        if citing_opinion is not None:
            params['citing_opinion'] = citing_opinion
        if cited_opinion is not None:
            params['cited_opinion'] = cited_opinion
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, citation_id: int) -> Dict[str, Any]:
        """
        Get a specific citation relationship by ID.
        
        Args:
            citation_id: Citation relationship ID
        
        Returns:
            Citation relationship data
        """
        return self.client.get(f"{self.endpoint}{citation_id}/")
    
    def paginate(
        self,
        citing_opinion: Optional[int] = None,
        cited_opinion: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated citation relationships.
        
        Args:
            citing_opinion: Filter by citing opinion ID
            cited_opinion: Filter by cited opinion ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for citation relationships
        """
        params = {}
        if citing_opinion is not None:
            params['citing_opinion'] = citing_opinion
        if cited_opinion is not None:
            params['cited_opinion'] = cited_opinion
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)

File: php/src/CourtListener/Api/OpinionsCited.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * OpinionsCited API endpoint
 */
class OpinionsCited extends BaseApi
{
    protected string $endpoint = 'opinions-cited/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List opinions cited with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listOpinionsCited(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific opinions cited by ID
     *
     * @param int|string $id OpinionsCited ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getOpinionsCited(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search opinions cited
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchOpinionsCited(array $params = []) {
        return $this->search($params);
    }
}
